File: task1/task1b/Packaging.cpp
#include "Packaging.h"
#include <iostream>

Packaging::Packaging(){}

Packaging::Packaging(int ID, double cost, const string& mat, int cap): packagingID(ID), packagingCost(cost), material(mat), capacity(cap){}

void Packaging::display_package_details(){
    cout << "Packaging ID: " << getID() << endl; 
    cout << "Packaging Cost: " << getCost() << endl;
    cout << "Packaging Material: " << getMaterial() << endl;
    cout << "Packaging Capacity: " << getCapacity() << endl;
}

// Setters
void Packaging::setID(int ID){
    packagingID = ID;
}

void Packaging::setCost(double cost){
    packagingCost = cost;
}

void Packaging::setMaterial(const string& mat){
    material = mat;
}

void Packaging::setCapacity(int cap){
    capacity = cap;
}

// Getters
int Packaging::getID() const{
    return packagingID;
}

double Packaging::getCost() const{
    return packagingCost;
}

string Packaging::getMaterial() const{
    return material;
}

int Packaging::getCapacity() const{
    return capacity;
}

File: task2/task2/src/main/java/com/task2/Packaging.java
package com.task2;

import java.io.Serializable;

public class Packaging implements Serializable{
    private static final long serialVersionUID = 1L;

    protected int packagingID;
    protected double packagingCost;
    protected String material;
    protected int capacity;

    public Packaging(){
        
    }

    public Packaging(int id, double cost, String mat, int cap){
        packagingID = id;
        packagingCost = cost;
        material = mat;
        capacity = cap;
    }

    public void display_package_details(){
        System.out.println("Packaging ID: "+getID());
        System.out.println("Packaging Cost: "+getCost());
        System.out.println("Packaging Material: "+getMaterial());
        System.out.println("Packaging Capacity: "+getCapacity());
    }

    // Setters
    public void setID(int id){
        packagingID = id;
    }

    public void setCost(double cost){
        packagingCost = cost;
    }

    public void setMaterial(String mat){
        material = mat;
    }

    public void setCapacity(int cap){
        capacity = cap;
    }

    // Getters
    public int getID(){
        return packagingID;
    }

    public double getCost(){
        return packagingCost;
    }

    public String getMaterial(){
        return material;
    }

    public int getCapacity(){
        return capacity;
    }
}
